### PR TITLE
8 website sale autopay website_sale_order_company

### PR DESCRIPTION
**Pull request by ilyasProgrammer**

### CODE REVIEW

File: website_sale_autopay/website_sale_autopay_models.py [hunk 1]
```diff
@@ -1,24 +1,33 @@
 # -*- coding: utf-8 -*-
 from openerp import fields
 from openerp import models
+import logging
 import re
 
+_logger = logging.getLogger(__name__)
 
 class PaymentAcquirer(models.Model):
```
Comment (yelizariev) [inline website_sale_autopay/website_sale_autopay_models.py]: fix class name

File: website_sale_autopay/website_sale_autopay_models.py [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
-    _inherit = 'payment.acquirer'
+    _inherit = 'account.journal'
 
-    journal_id = fields.Many2one('account.journal', 'Payment method', help='This journal is used to auto pay invoice when online payment is received')
+    # journal_id = fields.Many2one('account.journal', 'Payment method', help='This journal is used to auto pay invoice when online payment is received')
+    payment_acquirer_id = fields.Many2one('payment.acquirer', 'Payment acquirer', required=True)
```
Comment (yelizariev) [inline website_sale_autopay/website_sale_autopay_models.py]: why required?

File: website_sale_autopay/website_sale_autopay_models.py [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
 
 
-class sale_order(models.Model):
-
+class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
     def action_button_confirm(self, cr, uid, ids, context=None):
```
Comment (yelizariev) [inline website_sale_autopay/website_sale_autopay_models.py]: use new api

**Comment by yelizariev** [comment]
fix lint errors:

```
website_sale_autopay/website_sale_autopay_models.py:9:1: E302 expected 2 blank lines, found 1
website_sale_autopay/website_sale_autopay_models.py:25:72: E128 continuation line under-indented for visual indent
```

**Comment by yelizariev** [comment]
squash it

**Comment by yelizariev** [comment]
:-1:

**Comment by yelizariev** [comment]
:-1:

**Comment by ilyasProgrammer** [comment]
:fu:

**Comment by yelizariev** [comment]
Don't remove commit eb2afbd. Add new one instead

File: website_sale_autopay/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+==================================
+ Auto invoice creation on payment
+==================================
+
+When payment is processed thru some provider (acquirer), such as PayPal,
+the invoice (Account voucher) is created with corresponding accounting entries.
```
Comment (yelizariev) [inline website_sale_autopay/doc/index.rst]: https://odoo-development.readthedocs.io/en/latest/dev/docs/doc-files-understanding.html#index-rst

**Comment by yelizariev** [comment]
fix lint error
************* Module website_sale_autopay.website_sale_autopay_models
website_sale_autopay/website_sale_autopay_models.py:79: [E1601(print-statement), SaleOrder.action_button_confirm] print statement used

File: website_sale_autopay/doc/index.rst [hunk 5]
```diff
@@ -2,5 +2,21 @@
...
+============
+
+* `Install <https://odoo-development.readthedocs.io/en/latest/odoo/usage/install-module.html>`__ this module in a usual way
+
+Configuration
+=============
+
+
+* `Enable technical features <https://odoo-development.readthedocs.io/en/latest/odoo/usage/technical-features.html>`__
+* Set acquirer for every journal to be used for eCommerce payments registration.
```
Comment (yelizariev) [inline website_sale_autopay/doc/index.rst]: Напиши подробно в какое меню нужно заходить

File: website_sale_autopay/doc/index.rst [hunk 6]
```diff
@@ -2,5 +2,21 @@
...
+
+Usage
+=====
+
 When payment is processed thru some provider (acquirer), such as PayPal,
 the invoice (Account voucher) is created with corresponding accounting entries.
+You can look up created vouchers to see that payments.
```
Comment (yelizariev) [inline website_sale_autopay/doc/index.rst]: В каком меню

**Comment by yelizariev** [comment]
fix doc and make single commit for doc updates (don't touch eb2afbd )

File: website_sale_autopay/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+* `Enable technical features <https://odoo-development.readthedocs.io/en/latest/odoo/usage/technical-features.html>`__
+* Set acquirer for every journal to be used for eCommerce payments registration.
+
+Usage
+=====
+
+When payment is processed thru some provider (acquirer), such as PayPal,
+the invoice (Account voucher) is created with corresponding accounting entries.
+You can look up created vouchers to see that payments.
```
Comment (yelizariev) [inline website_sale_autopay/doc/index.rst]: В каком меню напиши

File: website_sale_autopay/doc/index.rst [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    * Set ``Payment acquirer`` for every journal to be used for eCommerce payments registration.
+
+
+Usage
+=====
+
+When payment is processed thru some provider (acquirer), such as PayPal,
+the invoice (Account voucher) is created with corresponding accounting entries.
+You can look up created vouchers in ``Accounting / Journal Entries / Journal Vouchers``
+(using user with ``Technical Features`` enabled and with Accountant rights) to see those payments.
```
Comment (yelizariev) [inline website_sale_autopay/doc/index.rst]: Use link here

  `Technical Features <https://odoo-development.readthedocs.io/en/latest/odoo/usage/technical-features.html>`__

**Comment by yelizariev** [comment]
docs are ok

File: website_sale_autopay/website_sale_autopay_models.py [hunk 9]
```diff
@@ -1,24 +1,29 @@
...
-        super(sale_order, self).action_button_confirm(cr, uid, ids, context=context)
+        super(SaleOrder, self).action_button_confirm(cr, uid, ids, context=context)
         r = self.browse(cr, uid, ids[0], context=context)
         if r.payment_tx_id and r.payment_tx_id.state == 'done' and r.payment_acquirer_id:
             journal_id = r.payment_acquirer_id.journal_id.id or self.pool['account.invoice'].default_get(cr, uid, ['journal_id'], context=context)['journal_id']
+            if not journal_id:
+                _logger.warning("No journal for payment acquirer %s.", r.payment_acquirer_id)
```
Comment (yelizariev) [inline website_sale_autopay/website_sale_autopay_models.py]: :+1:

**Pull request closed without merging**